pkg/bot/mattermost.go

```diff
@@ -285,3 +322,94 @@ func (b MMBot) listen(ctx context.Context) {
 		}
 	}
 }
+
+// SendEvent sends event notification to Mattermost
+func (b *Mattermost) SendEvent(ctx context.Context, event events.Event) error {
+	if !b.notify {
+		b.log.Info("Notifications are disabled. Skipping event...")
+		return nil
+	}
+
+	b.log.Debugf(">> Sending to Mattermost: %+v", event)
+
+	var fields []*model.SlackAttachmentField
+
+	switch b.Notification.Type {
+	case config.LongNotification:
+		fields = b.longNotification(event)
+	case config.ShortNotification:
+		fallthrough
+
+	default:
+		// set missing cluster name to event object
+		fields = b.shortNotification(event)
+	}
+
+	attachment := []*model.SlackAttachment{
+		{
+			Color:     attachmentColor[event.Level],
+			Title:     event.Title,
+			Fields:    fields,
+			Footer:    "BotKube",
+			Timestamp: json.Number(strconv.FormatInt(event.TimeStamp.Unix(), 10)),
+		},
+	}
+
+	targetChannel := event.Channel
+	if targetChannel == "" {
+		// empty value in event.channel sends notifications to default channel.
+		targetChannel = b.ChannelID
+	}
+	isDefaultChannel := targetChannel == b.ChannelID
+
+	post := &model.Post{
+		Props: map[string]interface{}{
+			"attachments": attachment,
+		},
+		ChannelId: targetChannel,
+	}
+
+	_, resp := b.APIClient.CreatePost(post)
+	if resp.Error != nil {
+		createPostWrappedErr := fmt.Errorf("while posting message to channel %q: %w", targetChannel, resp.Error)
+
+		if isDefaultChannel {
+			return createPostWrappedErr
+		}
+
+		// fallback to default channel
+
+		// send error message to default channel
+		msg := fmt.Sprintf("Unable to send message to ChannelName `%s`: `%s`\n```add Botkube app to the ChannelName %s\nMissed events follows below:```", targetChannel, resp.Error, targetChannel)
```

Comment:
```suggestion
		msg := fmt.Sprintf("Unable to send message to channel `%s`: `%s`\n```add Botkube app to the channel %s\nMissed events follows below:```", targetChannel, resp.Error, targetChannel)
```

yep, the refactor tool can be harmful 😄

Reply:
😞 Yep